query-planner-js/src/composedSchema/removeInaccessibleElements.ts

```diff
@@ -40,38 +41,41 @@ export function removeInaccessibleElements(
 
   removeRootTypesIfNeeded();
 
-  return transformSchema(schema, (type) => {
-    // Remove the type.
-    if (typesToRemove.has(type)) return null;
-
-    if (isObjectType(type)) {
-      const typeConfig = type.toConfig();
-
-      return new GraphQLObjectType({
-        ...typeConfig,
-        fields: removeInaccessibleFields(type, typeConfig.fields),
-        interfaces: removeInaccessibleTypes(typeConfig.interfaces)
-      });
-    } else if (isInterfaceType(type)) {
-      const typeConfig = type.toConfig();
-
-      return new GraphQLInterfaceType({
-        ...typeConfig,
-        fields: removeInaccessibleFields(type, typeConfig.fields),
-        interfaces: removeInaccessibleTypes(typeConfig.interfaces)
-      });
-    } else if (isUnionType(type)) {
-      const typeConfig = type.toConfig();
-
-      return new GraphQLUnionType({
-        ...typeConfig,
-        types: removeInaccessibleTypes(typeConfig.types)
-      });
-    } else {
-      // Keep the type as is.
-      return undefined;
-    }
-  });
+  return {
+    schema: transformSchema(schema, (type) => {
+      // Remove the type.
+      if (typesToRemove.has(type)) return null;
+
+      if (isObjectType(type)) {
+        const typeConfig = type.toConfig();
+
+        return new GraphQLObjectType({
+          ...typeConfig,
+          fields: removeInaccessibleFields(type, typeConfig.fields),
+          interfaces: removeInaccessibleTypes(typeConfig.interfaces),
+        });
+      } else if (isInterfaceType(type)) {
+        const typeConfig = type.toConfig();
+
+        return new GraphQLInterfaceType({
+          ...typeConfig,
+          fields: removeInaccessibleFields(type, typeConfig.fields),
+          interfaces: removeInaccessibleTypes(typeConfig.interfaces),
+        });
+      } else if (isUnionType(type)) {
+        const typeConfig = type.toConfig();
+
+        return new GraphQLUnionType({
+          ...typeConfig,
+          types: removeInaccessibleTypes(typeConfig.types),
+        });
+      } else {
+        // Keep the type as is.
+        return undefined;
+      }
+    }),
```

Comment:
nit: I think an assignment to a separate `filteredSchema` with the contents of `transformSchema`, and then just `{ schema: filteredSchema, removedTypes: typesToRemove }` could be easier to read and inspect.

Reply:
outdated